pilot/pkg/xds/eds.go

```diff
@@ -165,9 +168,37 @@ func (s *DiscoveryServer) edsCacheUpdate(shard model.ShardKey, hostname string,
 	// For existing endpoints, we need to do full push if service accounts change.
 	if saUpdated {
 		log.Infof("Full push, service accounts changed, %v", hostname)
-		fullPush = true
+		return true, true
+	}
+	if created {
+		return true, true
+	}
+	// Check if new Endpoints are ready to be pushed. This check
+	// will ensure that if a new pod comes with a non ready endpoint,
+	// we do not unnecessarily push that config to Envoy.
+	for _, nie := range istioEndpoints {
+		exists := false
+		for _, oie := range oldIstioEndpoints {
```

Comment:
O(N^2) check may be problematic here? we can have huge numbers

Reply:
I thought about it but could not find a better way. Let me see.

Reply:
I found a better way - will fix.